src/jace/stages.py

```diff
@@ -227,20 +234,23 @@ class JaCeLowered(tcache.CachingStage["JaCeCompiled"], Generic[_R]):
     _translated_sdfg: tjsdfg.TranslatedJaxprSDFG
     _out_tree: jax_tree.PyTreeDef
     _jaxpr: jax_core.ClosedJaxpr
+    _device: dace.DeviceType
 
     def __init__(
         self,
         tsdfg: tjsdfg.TranslatedJaxprSDFG,
         out_tree: jax_tree.PyTreeDef,
         jaxpr: jax_core.ClosedJaxpr,
+        device: str | dace.DeviceType,
     ) -> None:
         super().__init__()
         self._translated_sdfg = tsdfg
         self._out_tree = out_tree
         self._jaxpr = jaxpr
+        self._device = util.parse_backend_jit_option(device)
```

Comment:
This is a matter of style but I'd rather exclude checking the input type inside the parser function: 
```suggestion
        self._device = device if isinstance(str, dace.DeviceType) else util.parse_backend_jit_option(device)
```

Reply:
I deleted that line.
However, I think this should be put inside the parse function.
As you have to repeat it everytime you use it.